### PR TITLE
improve streams actor integration docs, #21698

### PR DESCRIPTION
**Pull request by patriknw**
- mapAsync + ask should be the first choice
- add missing Source.queue
- prefer actorRefWithAck
- move ActorPublisher and ActorSubscriber to the end
  with additional warning

### CODE REVIEW

**Comment by patriknw** [comment]
Refs #21698

File: akka-docs/rst/java/stream/stream-integrations.rst [hunk 1]
```diff
@@ -7,138 +7,109 @@ Integration
...
+as reply from the actor.
 
-Here is an example of such an actor. It dispatches incoming jobs to the attached subscriber:
+If the ``ask`` fails due to timeout the stream will be completed with
+``TimeoutException`` failure. If that is not desired outcome you can use ``recover`` 
+on the ``ask`` :class:`CompletionStage`.
 
-.. includecode:: ../code/docs/stream/ActorPublisherDocTest.java#job-manager
+If you don't care about the replies you can use ``Sink.ignore`` after the ``mapAsync``
+stage and then actor is effectively a sink of the stream.
```
Comment (johanandren) [inline akka-docs/rst/java/stream/stream-integrations.rst]: Maybe clarify that it is when you don't care but still want request-response to backpressure? I first read this and thought "why would I use ask if I don't care about the reply?".

**Review by johanandren — APPROVED** [review]
LGTM

File: akka-docs/rst/java/stream/stream-integrations.rst (hunk 1 above)
Comment (patriknw) [inline akka-docs/rst/java/stream/stream-integrations.rst]: good point, fixed b0b7387

File: akka-docs/rst/java/stream/stream-integrations.rst [hunk 2]
```diff
@@ -7,138 +7,110 @@ Integration
...
+The stream can be completed with failure by sending ``akka.actor.Status.Failure``
+as reply from the actor.
 
-Here is an example of such an actor. It dispatches incoming jobs to the attached subscriber:
+If the ``ask`` fails due to timeout the stream will be completed with
+``TimeoutException`` failure. If that is not desired outcome you can use ``recover`` 
+on the ``ask`` :class:`CompletionStage`.
 
-.. includecode:: ../code/docs/stream/ActorPublisherDocTest.java#job-manager
+If you don't care about the reply values and only use them back-pressure signals you
```
Comment (johanandren) [inline akka-docs/rst/java/stream/stream-integrations.rst]: "only use them _as_ back-pressure signals"

File: akka-docs/rst/java/stream/stream-integrations.rst [hunk 3]
```diff
@@ -7,138 +7,110 @@ Integration
...
+the stream). The elements will be buffered until the stream can process them. You can ``offer`` elements to 
+the queue and they will be emitted to the stream if there is demand from downstream, otherwise they will 
+be buffered until request for demand is received. 
 
-This is how it can be used as input :class:`Source` to a :class:`Flow`:
+Use overflow strategy ``akka.stream.OverflowStrategy.backpressure`` to avoid dropping of elements if the 
+buffer is full.
 
-.. includecode:: ../code/docs/stream/ActorPublisherDocTest.java#actor-publisher-usage
+``SourceQueue.offer`` returns ``CompletionStage<StreamCallbackStatus<Boolean>>`` which completes with ``Success(true)``
```
Comment (ktoso) [inline akka-docs/rst/java/stream/stream-integrations.rst]: The javadoc seems to be incorrect on those (!) - there is no `StreamCallbackStatus` in the entire project, it seems that `QueueOfferResult` is meant instead?

File: akka-docs/rst/java/stream/stream-integrations.rst [hunk 4]
```diff
@@ -438,3 +410,101 @@ passing a factory function that will create the :class:`Processor` instances:
...
+
+For historical reasons the :class:`ActorPublisher` and :class:`ActorSubscriber` traits are
+provided to support implementing Reactive Streams :class:`Publisher` and :class:`Subscriber` with
+an :class:`Actor`.
+
+These can be consumed by other Reactive Stream libraries or used as an Akka Streams :class:`Source` or :class:`Sink`.
+
+.. warning::
+
+  :class:`ActorPublisher` and :class:`ActorSubscriber` will probably be deprecated in future versions of Akka.
```
Comment (ktoso) [inline akka-docs/rst/java/stream/stream-integrations.rst]: good 👍

**Review by ktoso — CHANGES_REQUESTED** [review]
The non existing type needs to be fixed here ( https://github.com/akka/akka/pull/21700#discussion_r84068183 ) and in javadoc, otherwise very good improvement 👍

**Review by drewhk — APPROVED** [review]

**Comment by patriknw** [comment]
@ktoso fixed the wrong `StreamCallbackStatus`, and found missing java api: 1a102f7

**Review by ktoso — APPROVED** [review]

**Pull request merged**